datasets/liar/liar.py

```diff
@@ -0,0 +1,130 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""LIAR is a dataset for fake news detection with annotated claims."""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import os
+
+import datasets
+
+
+_CITATION = """\
+@inproceedings{wang-2017-liar,
+title = "{``}Liar, Liar Pants on Fire{''}: A New Benchmark Dataset for Fake News Detection",
+author = "Wang, William Yang",
+booktitle = "Proceedings of the 55th Annual Meeting of the Association for Computational Linguistics (Volume 2: Short Papers)",
+month = jul,
+year = "2017",
+address = "Vancouver, Canada",
+publisher = "Association for Computational Linguistics",
+url = "https://www.aclweb.org/anthology/P17-2067",
+doi = "10.18653/v1/P17-2067",
+pages = "422--426",
+abstract = "Automatic fake news detection is a challenging problem in deception detection, and it has tremendous real-world political and social impacts. However, statistical approaches to combating fake news has been dramatically limited by the lack of labeled benchmark datasets. In this paper, we present LIAR: a new, publicly available dataset for fake news detection. We collected a decade-long, 12.8K manually labeled short statements in various contexts from PolitiFact.com, which provides detailed analysis report and links to source documents for each case. This dataset can be used for fact-checking research as well. Notably, this new dataset is an order of magnitude larger than previously largest public fake news datasets of similar type. Empirically, we investigate automatic fake news detection based on surface-level linguistic patterns. We have designed a novel, hybrid convolutional neural network to integrate meta-data with text. We show that this hybrid approach can improve a text-only deep learning model.",
+}
+"""
+
+_DESCRIPTION = """\
+LIAR is a dataset for fake news detection with 12.8K human labeled short statements from politifact.com's API, and each statement is evaluated by a politifact.com editor for its truthfulness. The distribution of labels in the LIAR dataset is relatively well-balanced: except for 1,050 pants-fire cases, the instances for all other labels range from 2,063 to 2,638. In each case, the labeler provides a lengthy analysis report to ground each judgment.
+"""
+
+_HOMEPAGE = "https://www.aclweb.org/anthology/P17-2067"
+
+_LICENSE = "Unknown"
+
+_URL = "https://www.cs.ucsb.edu/~william/data/liar_dataset.zip"
+
+
+class Liar(datasets.GeneratorBasedBuilder):
+    """LIAR is a dataset for fake news detection with annotated claims."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "label": datasets.Value("string"),
```

Comment:
could you use the ClassLabel feature type here for the `label` field ?
for example
```
"label": datasets.Classlabel(names=["false", "half-true", etc.])
```

you'll need to regenerate the dataset_infos.json file after this change